homeassistant/components/geniushub/climate.py

```diff
@@ -26,27 +27,49 @@
 GH_ZONES = ["radiator", "wet underfloor"]
 
 
-async def async_setup_platform(
+def _do_setup(
     hass: HomeAssistant,
-    config: ConfigType,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Genius Hub climate entities."""
-    if discovery_info is None:
-        return
-
     broker = hass.data[DOMAIN]["broker"]
 
     async_add_entities(
         [
             GeniusClimateZone(broker, z)
             for z in broker.client.zone_objs
             if z.data.get("type") in GH_ZONES
-        ]
+        ],
+        update_before_add=True,
     )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Genius Hub climate entities."""
+    config = hass.data[DOMAIN][entry.entry_id]
+    # Update our config to include new repos and remove those that have been removed.
+    if entry.options:
+        config.update(entry.options)
+
+    _do_setup(hass, async_add_entities)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Genius Hub climate entities."""
+    if discovery_info is None:
+        return
+
+    _do_setup(hass, async_add_entities)
+
```

Comment:
All YAML is imported, so this isn't needed

Reply:
Done.